front/lib/api/assistant/global_agents.ts

```diff
@@ -161,6 +164,178 @@ async function _getSlackGlobalAgent(
   };
 }
 
+async function _getGoogleDriveGlobalAgent(
+  auth: Authenticator
+): Promise<AgentConfigurationType | null> {
+  const owner = auth.workspace();
```

Comment:
We need to check it's not null for the call to `prodAPICredentialsForOwner` I believe

Reply:
True, ok then. Was wondering whether that part could be dried up  since it's the same for all datasource-based agents

But not worth it at this time I think